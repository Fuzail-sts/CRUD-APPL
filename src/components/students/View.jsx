import React from "react";
import {
  Typography,
  Box,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableContainer,
  Button,
} from "@material-ui/core";
import { useParams } from "react-router";
import { useState, useEffect } from "react";
import axios from "axios";
import { Link } from "react-router-dom";
const View = () => {
  const { id } = useParams();
  console.log(id);
  const [viewList, setViewList] = useState([]);
  useEffect(() => {
    getStudentDataList();
  }, []);
  const getStudentDataList = () => {
    axios
      .get(`http://localhost:3333/students/${id}`)
      .then((response) => {
        setViewList(response.data);
      })
      .catch((error) => {
        console.log("something went wrong");
      });
  };
  return (
    <>
      <Box textAlign="center" mt={6}>
        <Typography variant="h4">STUDENT DETAIL</Typography>
      </Box>
      <TableContainer>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell align="center">ID</TableCell>
              <TableCell align="center">NAME</TableCell>
              <TableCell align="center">EMAIL</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
              <TableCell align="center">{viewList.id}</TableCell>
              <TableCell align="center">{viewList.name}</TableCell>
              <TableCell align="center">{viewList.email}</TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </TableContainer>
      <Box textAlign="center">
        <Link to="/">
          <Button variant="contained">BACK TO HOME</Button>
        </Link>
      </Box>
    </>
  );
};

export default View;
