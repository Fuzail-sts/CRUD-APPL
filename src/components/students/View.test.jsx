import React from "react";
import   renderer  from "react-test-renderer";
import View from "./View";
import { BrowserRouter, Route } from "react-router-dom";

describe("View", () => {
  it("snapShot", async () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Route>
            <View />
          </Route>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
