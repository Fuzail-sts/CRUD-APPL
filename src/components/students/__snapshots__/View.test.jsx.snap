// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View snapShot 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      STUDENT DETAIL
    </h4>
  </div>,
  <div
    className="MuiTableContainer-root"
  >
    <table
      className="MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            NAME
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
            scope="col"
          >
            EMAIL
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
          />
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          BACK TO HOME
        </span>
      </button>
    </a>
  </div>,
]
`;
